src/components/datasets/card/index.tsx

```diff
@@ -28,75 +27,46 @@ const DatasetCard: FC<DatasetCardProps> = ({
   author,
   gs_style: legendStyles,
   range,
-  autoPlay = false,
-  defaultActive,
+  defaultActive = false,
 }) => {
   const [layers, setLayers] = useSyncLayersSettings();
-  const [compareLayers, setCompareLayers] = useSyncCompareLayersSettings();
+  const [, setCompareLayers] = useSyncCompareLayersSettings();
+
+  // isActive is based on the url
   const layerId = layers?.[0]?.id;
-  const layerOpacity = layers?.[0]?.opacity;
-  const layerDate = layers?.[0]?.date;
-  const active = !!layerId || defaultActive;
-  const [isActive, setIsActive] = useState<boolean>(active);
-  const [isFirstRender, setIsFirstRender] = useState<boolean>(true);
-  const opacity = layerOpacity ?? 1;
-  const compareId = compareLayers?.[0]?.id;
-
-  // activates first layer at first render is there is nothing on the url
-  useEffect(() => {
-    if (isActive && !layerId && isFirstRender) {
-      void setLayers([
-        {
-          opacity,
-          date: range?.[0]?.value,
-          id: layerId || id,
-        },
-      ]);
-    }
-    setIsFirstRender(false);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  const isActive = useMemo(() => layerId === id, [id, layerId]);
 
-  // activates layer on url
-  useEffect(() => {
-    if (layerId) {
+  /**
+   * Handle click on the toggle button
+   */
+  const handleToggleLayer = useCallback(() => {
+    if (!isActive) {
       void setLayers([
         {
-          opacity,
-          date: layerDate,
-          id: layerId,
+          opacity: 1,
+          date: range?.[0]?.value,
+          id,
         },
       ]);
```

Comment:
```suggestion
if (!range) void setCompareLayers(null);
      ]);
```
so if the compare is open and we moved to a layer without range, the URL gets clear nd don't preserve the compare option